File: mobile/ios/packages/town-crier-domain/Sources/ValueObjects/ApplicationPage.swift
/// One page of watch-zone planning applications plus the opaque continuation
/// token for the next page. `nextCursor` is `nil` on the last page, which ends
/// the list's infinite-scroll loop (GH#682 slice 1).
public struct ApplicationPage: Sendable, Equatable {
  public let applications: [PlanningApplication]
  public let nextCursor: String?

  public init(applications: [PlanningApplication], nextCursor: String?) {
    self.applications = applications
    self.nextCursor = nextCursor
  }
}

File: mobile/android/domain/src/main/kotlin/uk/towncrierapp/domain/applications/ApplicationPage.kt
package uk.towncrierapp.domain.applications

/** One page of a zone's application list. A `null` [nextCursor] means this was the last page. Port of iOS `ApplicationPage` (GH#775). */
public data class ApplicationPage(
    public val applications: List<PlanningApplication>,
    public val nextCursor: String? = null,
)
